server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -367,128 +298,68 @@ public TimeValue getRerouteInterval() {
-     */
-    private static ByteSizeValue thresholdBytesFromWatermark(String watermark, String settingName, boolean lenient) {
-        try {
-            return ByteSizeValue.parseBytesSizeValue(watermark, settingName);
-        } catch (ElasticsearchParseException ex) {
-            // NOTE: this is not end-user leniency, since up above we check that it's a valid byte or percentage, and then store the two
-            // cases separately
-            if (lenient) {
-                return ByteSizeValue.ZERO;
-            }
-            throw ex;
-        }
+    public String describeFloodStageThreshold(ByteSizeValue total, boolean includeSettingKey) {
+        return describeThreshold(
+            total,
+            floodStageWatermark,
+            ByteSizeValue.ofBytes(-1),
+            includeSettingKey,
+            CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_WATERMARK_SETTING.getKey(),
+            null
+        );
     }
 
-    /**
-     * Checks if a watermark string is a valid percentage or byte size value,
-     * @return the watermark value given
-     */
-    private static String validWatermarkSetting(String watermark, String settingName) {
-        if (definitelyNotPercentage(watermark)) {
-            // short circuit to save expensive exception on obvious byte size value below
-            ByteSizeValue.parseBytesSizeValue(watermark, settingName);
-            return watermark;
-        }
-        try {
-            RatioValue.parseRatioValue(watermark);
-        } catch (ElasticsearchParseException e) {
-            try {
-                ByteSizeValue.parseBytesSizeValue(watermark, settingName);
-            } catch (ElasticsearchParseException ex) {
-                ex.addSuppressed(e);
-                throw ex;
-            }
-        }
-        return watermark;
+    public String describeFrozenFloodStageThreshold(ByteSizeValue total, boolean includeSettingKey) {
+        return describeThreshold(
+            total,
+            frozenFloodStageWatermark,
+            frozenFloodStageMaxHeadroom,
+            includeSettingKey,
+            CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_WATERMARK_SETTING.getKey(),
+            CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_MAX_HEADROOM_SETTING.getKey()
+        );
     }
 
-    // Checks that a value is definitely not a percentage by testing if it ends on `b` which implies that it is probably a byte size value
-    // instead. This is used to make setting validation skip attempting to parse a value as a percentage/ration for the settings in this
-    // class that accept either a byte size value. The main motivation of this method is to make tests faster. Some tests call this method
-    // frequently when starting up internal cluster nodes and using exception throwing and catching when trying to parse as a ratio as a
-    // means of identifying that a string is not a ratio is quite slow.
-    private static boolean definitelyNotPercentage(String value) {
```

Comment:
I think we should keep this optimization (i.e. implement it in `RelativeByteSizeValue`) since it's pretty powerful in tests.

Reply:
It's not used anymore and if not used, the checks will complain I believe. Should I keep it? And is there some qualifier I can add so checks do not complain?

Reply:
Sure _this_ method isn't used any more because we're parsing these values differently, but the new parser doesn't have this optimization and is therefore going to slow tests down. I think we should move this optimization to the new parsing code.

Reply:
Ah thanks for clarifying, I now understood from reading your comment and the comment above this function. Indeed let me see if I can move it somewhere, e.g., in RelativeByteSizeValue. I will come back to you.

Reply:
Hi @DaveCTurner , so after seeing the code in master, indeed this was used to speed up validations. But now with the RelativeByteValue class, we do not need to handle Strings in the validation, and the class already knows whether it's a percentage or absolute value. So there is no issue with validations.

However, I do see that this can speed up the parsing method of RelativeByteValue, so I moved this check and comment to org.elasticsearch.common.unit.RelativeByteSizeValue#parseRelativeByteSizeValue . Please check new commit and if it is not sufficient, comment and unresolve this conversation.
Thanks!